File: P102_BinaryTreeLevelOrderTraversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
        int k = 0;
        vector<vector<int>> res(1000);
        vector<int> temp;
        traversal(root, res, k);
        for(int i = 0; i < res.size(); i++){
            if(res[i].size() == 0) break;
            k++;
        }
        res.resize(k);
        return res;
    }
    void traversal(TreeNode* root, vector<vector<int>>& res, int k){
        if(!root) return;
        res[k].push_back(root->val);
        traversal(root->left, res, k+1);
        traversal(root->right, res, k+1);
    }
};

File: P102_Binary_Tree_Level_Order_Traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<List<Integer>> levelOrder(TreeNode root) {
        List<List<Integer>> L = new ArrayList<>();
        if(root == null) return L;
        Queue<TreeNode> q = new LinkedList<>();
        q.add(root);
        while(!q.isEmpty()){
            int n = q.size();
            List<Integer> temp = new ArrayList<>(n);
            for(int i = 0; i < n; i++){
                TreeNode node = q.poll();
                temp.add(node.val);
                
                if(node.left != null){
                    q.add(node.left);
                }
                if(node.right != null){
                    q.add(node.right);
                }
            }
            L.add(temp);
        }
        return L;
    }
    public void input (TreeNode root, List<List<Integer>> L, int k){
        if(root == null) return;
        
        List<Integer> tempArr = new ArrayList<>();
        
        if(L.size() <= k){
            L.add(new ArrayList<Integer>());
            tempArr = L.get(k);
            tempArr.add(root.val);
            
        }
        else{
            tempArr = L.get(k);
            tempArr.add(root.val);
        }
        input(root.left, L, k + 1);
        input(root.right, L, k + 1);
        return;
    }
}
